invenio/base/scripts/apache.py

```diff
@@ -17,188 +17,427 @@
+                elif self._resolved_dir in (Cfg.CANDIDATE_PATHS['apache'],
+                                            Cfg.CANDIDATE_PATHS['apache2']):
+                    pem_path = path.join(self._resolved_dir, 'server.pem')
+                    crt_path = path.join(self._resolved_dir, 'server.crt')
+                    key_path = path.join(self._resolved_dir, 'server.key')
+                elif self._resolved_dir == Cfg.CANDIDATE_PATHS['tls']:
+                    pem_path = ''
+                    crt_path = path.join(self._resolved_dir, 'localhost.crt')
+                    key_path = path.join(self._resolved_dir, 'localhost.key')
+                ssl = {
+                    'pem': pem_path,
+                    'crt': crt_path,
+                    'key': key_path,
+                }
+                # Drop files based on OS guesswork
+                if not ssl_pem_directive_needed:
+                    del ssl['pem']
+                else:
+                    del ssl['crt']
+                    del ssl['key']
+            else:
+                ssl = {key: val for key, val in (
+                    ('pem', app_cfg.get('CFG_APACHE_CERTIFICATE_PEM_FILE')),
+                    ('crt', app_cfg.get('CFG_APACHE_CERTIFICATE_CRT_FILE')),
+                    ('key', app_cfg.get('CFG_APACHE_CERTIFICATE_KEY_FILE')),
+                ) if val}
+
+            assert [i for i in ssl.values() if i], \
+                "Please provide at least one certificate file."
 
-        return [apc1, apc2]
+            return ssl
 
-    current_app.config.update(
-        SYS_PREFIX=sys.prefix,
-        CFG_RUNNING_AS_USER=pwd.getpwuid(os.getuid())[0],
-        CFG_WSGIDIR=os.path.abspath(
-            pkg_resources.resource_filename('invenio', '')))
+        @cached_property
+        def xsfpdirs(self):
+            """Get list of XPF directories.
 
-    apache_conf_dir = current_app.instance_path + os.sep + 'apache'
+            :returns: {
+                'web': str
+                'bibdocfile': str
+                'websubmit_storage': str
+                'tmp': str
+                }
+            :rtype: dict
+            """
+            xsfpdirs = {
+                'web': 'CFG_WEBDIR',
+                'bibdocfile': 'CFG_BIBDOCFILE_FILEDIR',
+                'websumit_storage': 'CFG_WEBSUMIT_STORAGEDIR',
+                'deposit_storage': 'CFG_DEPOSIT_STORAGEDIR',
+                'tmp': 'CFG_TMPDIR',
+            }
+            for key, val in xsfpdirs.items():
+                try:
+                    xsfpdirs[key] = app_cfg[val]
+                except KeyError:
+                    print(">>> Variable {} not set. Skipping.."
+                          .format(val), file=sys.stderr)
+                    del xsfpdirs[key]
+
+            xsfpdirs.update({
+                'attachfile': path.join(app_cfg['CFG_TMPDIR'], 'attachfile'),
+                'comments': path.join(app_cfg['CFG_DATADIR'], 'comments'),
```

Comment:
Woops. Very bad cherry picking followed by leaving my prompt like [1] this and then going home.
The variables are now back.

```
Enter passphrase for key '/home/gs-sis/.ssh/id_rsa': 
Enter passphrase for key '/home/gs-sis/.ssh/id_rsa': 
```